llama_index/schema.py

```diff
@@ -501,7 +502,31 @@ class IndexNode(TextNode):
     """
 
     index_id: str
-    obj: Any = Field(exclude=True)
+    obj: Any = None
+
+    def dict(self, **kwargs: Any) -> Dict[str, Any]:
+        from llama_index.storage.docstore.utils import doc_to_json
+
+        data = super().dict(**kwargs)
+
+        is_obj_serializable = False
+        try:
+            if self.obj is None:
+                data["obj"] = None
+            elif isinstance(self.obj, BaseNode):
+                data["obj"] = doc_to_json(self.obj)
+            elif isinstance(self.obj, BaseModel):
```

Comment:
not sure we could have more instance type support

Reply:
eh, we can add more in the future :)